Mage/src/main/java/mage/target/common/TargetCreatureOrPlaneswalkerAmount.java

```diff
@@ -11,12 +11,46 @@ public class TargetCreatureOrPlaneswalkerAmount extends TargetPermanentAmount {
     private static final FilterCreatureOrPlaneswalkerPermanent defaultFilter
             = new FilterCreatureOrPlaneswalkerPermanent("target creatures and/or planeswalkers");
 
+    /**
+     * <b>IMPORTANT</b>: Use more specific constructor if {@code amount} is not always the same number!<br>
+     * {@code minNumberOfTargets} defaults to zero for {@code amount} > 3, otherwise to one, in line with typical templating.<br>
+     * {@code maxNumberOfTargets} defaults to {@code amount}.<br>
+     * {@code filter} defaults to all creature and planeswalker permanents. ("target creatures and/or planeswalkers")
+     *
+     * @see TargetCreatureOrPlaneswalkerAmount#TargetCreatureOrPlaneswalkerAmount(int, int, int, FilterPermanent)
+     */
     public TargetCreatureOrPlaneswalkerAmount(int amount) {
-        super(amount, defaultFilter);
+        this(amount, defaultFilter);
     }
 
+    /**
+     * <b>IMPORTANT</b>: Use more specific constructor if {@code amount} is not always the same number!<br>
+     * {@code minNumberOfTargets} defaults to zero for {@code amount} > 3, otherwise to one, in line with typical templating.<br>
+     * {@code maxNumberOfTargets} defaults to {@code amount}.
+     *
+     * @see TargetCreatureOrPlaneswalkerAmount#TargetCreatureOrPlaneswalkerAmount(int, int, int, FilterPermanent)
+     */
     public TargetCreatureOrPlaneswalkerAmount(int amount, FilterPermanent filter) {
-        super(amount, filter);
+        this(amount, amount > 3 ? 0 : 1, amount, filter);
+    }
+
+    /**
+     * {@code filter} defaults to all creature and planeswalker permanents. ("target creatures and/or planeswalkers")
+     *
+     * @see TargetCreatureOrPlaneswalkerAmount#TargetCreatureOrPlaneswalkerAmount(int, int, int, FilterPermanent)
+     */
+    public TargetCreatureOrPlaneswalkerAmount(int amount, int minNumberOfTargets, int maxNumberOfTargets) {
+        this(amount, minNumberOfTargets, maxNumberOfTargets, defaultFilter);
+    }
+
+    /**
+     * @param amount             Amount of stuff (e.g. damage, counters) to distribute.
+     * @param minNumberOfTargets Minimum number of targets.
+     * @param maxNumberOfTargets Maximum number of targets. If no lower max is needed, set to {@code amount}.
+     * @param filter             Filter for creatures and/or planeswalkers that something will be distributed amongst.
+     */
+    public TargetCreatureOrPlaneswalkerAmount(int amount, int minNumberOfTargets, int maxNumberOfTargets, FilterPermanent filter) {
```

Comment:
Sorry, another nit:

The constructors in this class ought to take a FilterCreatureOrPlaneswalkerPermanent, rather than a FilterPermanent. Likewise for TargetCreaturePermanentAmount, should take a FilterCreaturePermanent.

Reply:
Good idea. I also added new constructors for `FilterControlledCreaturePermanent`, since many cards use it, and for `FilterControlledCreatureOrPlaneswalkerPermanent`, for future proofing.